source/Player.hx

```diff
@@ -187,6 +210,48 @@ import flixel.group.FlxGroup;
     btmBox.y = y + height - hitBoxHeight;
   }
 
+  //Override player.kill to drop any weilded items
+  override public function kill():Void 
+  {
+	  if (weilding){
+		dropCurrentEquip();
+	  }
+	  super.kill();
+	  new FlxTimer().start(2, resetLevel, 1); //This helps speed things up for debugging
+  }
+  
+  private function resetLevel(Timer:FlxTimer):Void
```

Comment:
Might poke @SamBumgardner about it.  I know it would require the extra check to put it in `PlayState`, but it's literally a State operation.